### DISCUSSION TITLE
alloc_iob function return NULL in few servers

### DISCUSSION THREAD

**Discussion by rmohamedshah** [ipxe/ipxe#667] [Q&A]
Recently I moved to latest iPXE source code with my oprom driver code and generated the image. I tried the booting with this changes on various server,  i am getting  failure in alloc_iob function in various server. In the same server, if i tried with the old image which have old ipXE source code (where used malloc_dma_offset inside the API alloc_iob) and its working fine. Please share me any pointers to debug this issue

**Comment by NiKiZe** [comment]
Use git bisect to figure out which commit is causing this. Also share your code?

**Comment by mcb30** [comment]
You will need to show your code before anyone is able to help you.

Please also note that `alloc_iob()` is expected to fail under low memory conditions, and drivers need to handle this gracefully.  If your driver does not handle a `NULL` return from `alloc_iob()` then you have a bug in your driver.
